doc/ci.rst

```diff
@@ -0,0 +1,308 @@
+======================
+Continuous Integration
+======================
+
+.. default-role:: code
+.. include:: rstcommon.rst
+
+.. contents::
+
+This document aims to be a comprehensive guide to using and developing the
+Continuation Integration (CI) system employed by |NimSkull|.
+
+Troubleshooting failures
+========================
+
+Failing tests
+-------------
+
+While "Run tester" log will give you the full detail on the test run, it can be
+very hard to find the failing test in the jungle of passing tests, so there are
+several integrations built into CI to make this easier:
+
+* Annotations are generated for test failures:
+  * If you are adding new tests in your PR, failure in any of them will be
+    annotated and you can find them directly in the "Files changed" view of the
+    PR
+
+* All test failures are isolated and printed in "Print all test errors" step.
+  By default Github Actions collapse this step so you'll have to expand it.
+
+"There are differences when building from source archive compared to building from git"
+---------------------------------------------------------------------------------------
+
+There are several reasons for why this happens:
+
+* The compiler is using information only available when building directly from
+  git, and the source archive generation routine is not updated to include this
+  information in the archive and/or the compiler build is not coded to use the
+  stored information.
+
+* A cosmic ray hit the CI test system and made it generate a slightly different
+  binary from the same source code. A re-run should clear the error in this case.
+
+To get more details about the error, open the failing workflow run and go to
+"Summary". Scroll down to the "Artifacts" section and look for one named
+"differences between binary from source archive and git". This contains an HTML
+report generated by ``diffoscope`` about the differences.
+
+If you need more information, the binaries that were compared can be found in
+the following artifacts:
+
+* "binaries from source archive"
+
+* "release binaries"
+
+Reproducibility test failures
+-----------------------------
+
+PRs to be merged are tested to make sure that the compiler build remains
+deterministic with regards to the environment. For more information, see
```

Comment:
```suggestion
deterministic with regards to the environment. To learn more about why reproducible builds matter and the concepts, see
```

Something like that helps indicate that it's general knowledge and not specific to us. This is not a hard suggestion.